Formula/spidermonkey@78.rb

```diff
@@ -0,0 +1,73 @@
+class SpidermonkeyAT78 < Formula
+  desc "JavaScript-C Engine"
+  homepage "https://spidermonkey.dev"
+  url "https://archive.mozilla.org/pub/firefox/releases/78.15.0esr/source/firefox-78.15.0esr.source.tar.xz"
+  version "78.15.0"
+  sha256 "a4438d84d95171a6d4fea9c9f02c2edbf0475a9c614d968ebe2eedc25a672151"
+  license "MPL-2.0"
+
+  depends_on "autoconf@2.13" => :build
+  depends_on "pkg-config" => :build
+  depends_on "python@3.9" => :build
+  depends_on "rust" => :build
+  # configure error: ld: unknown/unsupported architecture name for: -arch armv4t
+  depends_on arch: :x86_64
+  depends_on "icu4c"
+  depends_on "nspr"
+  depends_on "readline"
+
+  uses_from_macos "llvm" => :build
```

Comment:
Having `llvm` as a build-only dependency is a little weird. Is it because the build insists on using `clang`?

Reply:
I actually wasn't able to build on local Linux container so didn't get chance to check result.

I only saw it fail early on due to missing `llvm-*` binaries.
Arch does use LLVM for LTO, but I think compiling with GCC works.

Reply:
If it's using `llvm-*` binaries it's probably safer to use only the `llvm` toolchain instead of mixing. On the other hand, you might then get linkage with `libc++` (and yet still indirectly need GCC runtime libraries...)...

Reply:
Specifically, it is looking for `llvm-objdump`. Since GCC should work and Arch has runtime dependency on `gcc-libs` (https://github.com/archlinux/svntogit-packages/blob/packages/js78/trunk/PKGBUILD#L10), I think it is probably okay to mix here. Fedora also adds both compilers as build dependencies and then picks GCC.

---

```
@depends(host_c_compiler, c_compiler, bindgen_config_paths)
def llvm_objdump(host_c_compiler, c_compiler, bindgen_config_paths):
    clang = None
    for compiler in (host_c_compiler, c_compiler):
        if compiler and compiler.type == 'clang':
            clang = compiler.compiler
            break
        elif compiler and compiler.type == 'clang-cl':
            clang = os.path.join(os.path.dirname(compiler.compiler), 'clang')
            break

    if not clang and bindgen_config_paths:
        clang = bindgen_config_paths.clang_path
    llvm_objdump = 'llvm-objdump'
    if clang:
        out = check_cmd_output(clang, '--print-prog-name=llvm-objdump',
                               onerror=lambda: None)
        if out:
            llvm_objdump = out.rstrip()
    return (llvm_objdump,)
```

Reply:
From NixOS package, sounds like `llvm-objdump` is the only reason LLVM is necessary (https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/spidermonkey/91.nix#L35). Will add a comment in next attempt
```suggestion
  uses_from_macos "llvm" => :build # for llvm-objdump
```